components/style/values/animated/svg.rs

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! Animated types for CSS values in SVG
+
+use app_units::Au;
+use values::animated::{Animate, Procedure};
+use values::computed::{NonNegativeNumber, Number, Percentage};
+use values::computed::length::{CalcLengthOrPercentage, LengthOrPercentage};
+use values::computed::length::NonNegativeLengthOrPercentage;
+
+/// Stroke-* value support unit less value, so servo interpolate length value as number.
+#[derive(Clone, ComputeSquaredDistance, Copy, Debug, PartialEq, ToAnimatedZero)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+#[allow(missing_docs)]
+pub enum SvgNumberOrPercentageOrCalc {
+    Number(Number),
+    Percentage(Percentage),
+    Calc(CalcLengthOrPercentage),
+}
+
+impl From<NonNegativeLengthOrPercentage> for SvgNumberOrPercentageOrCalc {
+    fn from(lop: NonNegativeLengthOrPercentage) -> SvgNumberOrPercentageOrCalc {
+        lop.0.into()
+    }
+}
+
+impl From<NonNegativeNumber> for SvgNumberOrPercentageOrCalc {
+    fn from(num: NonNegativeNumber) -> SvgNumberOrPercentageOrCalc {
+        num.0.into()
+    }
+}
+
+impl From<LengthOrPercentage> for SvgNumberOrPercentageOrCalc {
+    fn from(lop: LengthOrPercentage) -> SvgNumberOrPercentageOrCalc {
+        match lop {
+            LengthOrPercentage::Length(len) => {
+                SvgNumberOrPercentageOrCalc::Number(len.to_f32_px())
+            },
+            LengthOrPercentage::Percentage(p) => {
+                SvgNumberOrPercentageOrCalc::Percentage(p)
+            },
+            LengthOrPercentage::Calc(calc) => {
+                SvgNumberOrPercentageOrCalc::Calc(calc.into())
+            },
+        }
+    }
+
+}
+
+impl From<Number> for SvgNumberOrPercentageOrCalc {
```

Comment:
Where is this conversion needed?

Reply:
These conversions are for `ToAnimated` of `SvgLengthOrPercentageOrNumber`. So I moved these conversions to `generics/svg.rs`, and commented.